aten/src/ATen/native/transformers/cuda/attention.cu

```diff
@@ -858,21 +807,26 @@ std::tuple<at::Tensor, at::Tensor> _efficient_attention_forward(
     const at::Tensor& query, // [b, seqlen, num_heads, K]
     const at::Tensor& key, // [b, seqlen, num_heads, K]
     const at::Tensor& value, // [b, seqlen, num_heads, Kv]
+    const c10::optional<at::Tensor>& bias, // [b, num_heads, seqlen, seqlen]
     // (Mode 1MHK only) [b+1]: cu_seqlens_q[b] contains the
     // position of the first query token for batch $b
-    const c10::optional<at::Tensor>& cu_seqlens_q,
-    // (Mode 1MHK only) [b+1]: cu_seqlens_k[b] contains the
+    const c10::optional<at::Tensor>& seqstart_q,
+    // (Mode 1MHK only) [b+1]: cu_seqlen_k[b] contains the
     // position of the first key token for batch $b
-    const c10::optional<at::Tensor>& cu_seqlens_k,
+    const c10::optional<at::Tensor>& seqstart_k,
     // (Mode 1MHK only) Maximum sequence length across batches
     const c10::optional<int64_t> max_seqlen_q_,
+    double dropout_p, // attention matrix dropout probability
+    int64_t custom_mask_type,
     bool compute_logsumexp,
-    bool causal,
-    c10::optional<double> scale) {
+    c10::optional<double> scale,
+    const c10::optional<at::Tensor>& causal_diagonal,
+    const c10::optional<at::Tensor>& seqlen_k) {
```

Comment:
Does this represent max_seqlen_k ? @danthe3rd

Reply:
`max_seqlen_k` is not used by the kernel.
`seqlen_k` is used when keys/values are padded (and typically this is useful when you have a fixed-size KV-cache during inference, and you want to avoid copying/growing tensors)